MBBSEmu/CPU/CPUCore.cs

```diff
@@ -1674,6 +1593,11 @@ private void Op_Retf()
         private void Op_Ret()
         {
             Registers.IP = Pop();
+
+            //Pop N bytes (N/2 words) that were Pushed before the CALL
+            if (_currentInstruction.Op0Kind == OpKind.Immediate16)
+                for (var i = 0; i < GetOperandValueUInt16(OpKind.Immediate16, EnumOperandType.Destination) / 2; i++)
+                    Pop();
```

Comment:
instead of looping through this, can't you just increment SP by the value and call it a day?